### ISSUE TITLE
Problems handling a `file-header` with multiple lines

### ISSUE THREAD

**Issue by DusterTheFirst** [palantir/tslint#4328]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: Version 3.1.6
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
/*!
 * hi
 * im cool
 */

import React from "react";
// ...
```

with `tslint.json` configuration:

```json
// ...
"file-header": [
    true,
    "hi\nim cool",
    "hi\nim cool"
]
// ...
```

#### Actual behavior
After running `Fix: missing file header` tslint errors with `missing file header`
![image](https://user-images.githubusercontent.com/14093962/49242502-ebd6ab80-f3d8-11e8-80cf-89655eb7a972.png)
![image](https://user-images.githubusercontent.com/14093962/49242528-fee97b80-f3d8-11e8-9d9e-73b60e45b53e.png)

#### Expected behavior
No error created from the header

#### Note
This header is a demo header in order to demonstrate the error. The [real header](https://gist.github.com/DusterTheFirst/f6b6d32c876b748fa4f2daaa722f53f8) I am trying to add has many more lines, but throws the same error.

**Comment by giladgray** [comment]
@DusterTheFirst this should be supported. try including the `!` in your config strings?

**Comment by DusterTheFirst** [comment]
```json
"file-header": [
    true,
    "!\nhi\nim cool",
    "hi\nim cool"
]
```
and
```json
"file-header": [
    true,
    "!hi\nim cool",
    "hi\nim cool"
]
```
Both do not work

**Comment by JoshuaKGoldberg** [comment]
This is amusing. Given the settings in the OP, with the VS code extension set to `--fix` on file save, every time you save the file a new copy of the file header is added.

```typescript
/*!
 * hi
 * im cool
 */

/*!
 * hi
 * im cool
 */

import React from "react";
// ...

**Comment by DusterTheFirst** [comment]
Yes, I might open a pr to fix/look into the bug, but for now all i can guess that is happening is that the regex matching does not account for multiple lines. I have seen no proof so far of multiline file-headers that work.

**Comment by DusterTheFirst** [comment]
What does your working tsconfig look like @giladgray?

**Comment by hubgit** [comment]
I was able to work around this with the following fixes to the header text:

```js
const matcher = licenseHeader
  .replace(/^/, '!\n') // add an exclamation mark and a newline at the start
  .replace(/[\\\/^$.*+?()[\]{}|]/g, '\\$&') // escape for regexp (lodash.escaperegexp)
  .replace(/\n/g, '\n \\* ?') // add leading asterisk to each line, possibly with an extra space
  .replace('YEAR', '\\d{4}') // add the year pattern
```

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!
